### PR TITLE
Ii 2115 pufa ticket parser

### PR DESCRIPTION
**Pull request by yqian10**

### CODE REVIEW

**Comment by ptmr3** [comment]
@labideli please review and approve when everything is ready for my review and merge, thank you

File: pufaAgent/getlogs_Pufa.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+import socket
+import pandas as pd
+import pytz
+from time import gmtime, strftime
+
+def setLoggerConfig():
+    # Get the root logger
+    logger = logging.getLogger(__name__)
+    # Have to set the root logger level, it defaults to logging.WARNING
+    logger.setLevel(logging.DEBUG)
```
Comment (moharnab123saikia) [inline pufaAgent/getlogs_Pufa.py]: set logging to info. We need to make it configurable later.

File: pufaAgent/getlogs_Pufa.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    loggingHandlerOut.setLevel(logging.DEBUG)
+    loggingHandlerOut.addFilter(LessThanFilter(logging.WARNING))
+    logger.addHandler(loggingHandlerOut)
+
+    loggingHandlerErr = logging.StreamHandler(sys.stderr)
+    loggingHandlerErr.setLevel(logging.WARNING)
+    logger.addHandler(loggingHandlerErr)
+    return logger
+
+class LessThanFilter(logging.Filter):
```
Comment (moharnab123saikia) [inline pufaAgent/getlogs_Pufa.py]: use underscore naming here and other applicable places

File: pufaAgent/getlogs_Pufa.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                      action="store", dest="MaxInTag", help="Max number of one tag can have")
+    (options, args) = parser.parse_args()
+
+    parameters = {}
+    if options.homepath is None:
+        parameters['homepath'] = os.getcwd()
+    else:
+        parameters['homepath'] = options.homepath
+    if options.serverUrl == None:
+        #parameters['serverUrl'] = 'http://stg.insightfinder.com'
```
Comment (moharnab123saikia) [inline pufaAgent/getlogs_Pufa.py]: set default host to https://app.insightfinder.com

File: pufaAgent/getlogs_Pufa.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+            if samplingInterval[-1:] == 's':
+                configVars['samplingInterval'] = float(float(samplingInterval[:-1]) / 60.0)
+            else:
+                configVars['samplingInterval'] = int(samplingInterval)
+    except IOError:
+        logger.error("config.ini file is missing")
+    return configVars
+
+
+def parseMessage():
```
Comment (moharnab123saikia) [inline pufaAgent/getlogs_Pufa.py]: This function is reading messages for a file. Use a suggestive name or add comments

**Comment by yqian10** [comment]
The code is fixed based on Morhanab's suggestion. It is ready to merge.

File: pufaAgent/getlogs_Pufa.py [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    else:
+        parameters['max_in_tag'] = int(options.max_in_tag)
+    return parameters
+
+
+def is_time_format(time_string, datetime_format):
+    """
+    Determines the validity of the input date-time string according to the given format
+    Parameters:
+    - `timeString` : datetime string to check validity
```
Comment (moharnab123saikia) [inline pufaAgent/getlogs_Pufa.py]: parameter name is wrong

File: pufaAgent/getlogs_Pufa.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+    - `temp_id` : datetime format to compare with
```
Comment (moharnab123saikia) [inline pufaAgent/getlogs_Pufa.py]: name is wrong

File: pufaAgent/getlogs_Pufa.py [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+    to_send_data_json = json.dumps(to_send_data_dict)
+
+    # send the data
+    post_url = parameters['server_url'] + "/customprojectrawdata"
+    response = requests.post(post_url, data=json.loads(to_send_data_json))
+    if response.status_code == 200:
+        logger.info(str(len(bytearray(to_send_data_json))) + " bytes of data are reported.")
+    else:
+        logger.info("Failed to send data.")
```
Comment (moharnab123saikia) [inline pufaAgent/getlogs_Pufa.py]: should be logger.error

File: pufaAgent/getlogs_Pufa.py [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+if __name__ == "__main__":
+    reload(sys)
+    sys.setdefaultencoding('utf-8')
+    logger = set_logger_config()
+    parameters = get_parameters()
+    config_vars = get_agent_config_vars()
+    try:
+        read_message()
+    except KeyboardInterrupt:
+        print "Interrupt from keyboard"
```
Comment (moharnab123saikia) [inline pufaAgent/getlogs_Pufa.py]: use logger. remove all print statements.

**Comment by moharnab123saikia** [comment]
I found a few small fixes. I am sleeping now. You can just change those and merge. The rest looks ok. Sorry about adding those comments later. Found them while I was verifying.

**Comment by yqian10** [comment]
Thanks for your review Moharnab. I am ready to merge.

**Pull request merged**